src/mesh/NineSlicePlane.js

```diff
@@ -0,0 +1,241 @@
+var DEFAULT_BORDER_SIZE= 10;
+
+var Plane = require('./Plane');
+
+/**
+ * The NineSlicePlane allows you to stretch a texture using 9-slice scaling. The corners will remain unscaled (useful
+ * for buttons with rounded corners for example) and the other areas will be scaled horizontally and or vertically
+ *  
+ *```js
+ * var Plane9 = new PIXI.NineSlicePlane(PIXI.Texture.fromImage("BoxWithRoundedCorners.png"), 15, 15, 15, 15);
+ *  ```
+ * <pre>
+ *      A                          B
+ *    +---+----------------------+---+
+ *  C | 1 |          2           | 3 |
+ *    +---+----------------------+---+
+ *    |   |                      |   |
+ *    | 4 |          5           | 6 |
+ *    |   |                      |   |
+ *    +---+----------------------+---+
+ *  D | 7 |          8           | 9 |
+ *    +---+----------------------+---+
+
+ *  When changing this objects width and/or height:
+ *     areas 1 3 7 and 9 will remain unscaled.
+ *     areas 2 and 8 will be stretched horizontally
+ *     areas 4 and 6 will be stretched vertically
+ *     area 5 will be stretched both horizontally and vertically
+ * </pre>
+ *
+ * @class
+ * @extends PIXI.mesh.Plane
+ * @memberof PIXI.mesh
+ * @param {PIXI.Texture} texture - The texture to use on the NineSlicePlane.
+ * @param {int} [leftWidth=10] size of the left vertical bar (A)
+ * @param {int} [topHeight=10] size of the top horizontal bar (C)
+ * @param {int} [rightWidth=10] size of the right vertical bar (B)
+ * @param {int} [bottomHeight=10] size of the bottom horizontal bar (D)
+ *
+ */
+function NineSlicePlane(texture, leftWidth, topHeight, rightWidth, bottomHeight)
+{
+	Plane.call(this, texture, 4, 4);
+
+	var uvs = this.uvs;
+	// right and bottom uv's are always 1
+	uvs[6] = uvs[14] = uvs[22] = uvs[30] = 1;
+	uvs[25] = uvs[27] = uvs[29] = uvs[31] = 1;
+
+	this._origWidth = texture.width;
+	this._origHeight = texture.height;
+	/**
+	 * The width of the NineSlicePlane, setting this will actually modify the vertices and UV's of this plane
+	 *
+	 * @member {number}
+	 * @memberof PIXI.NineSlicePlane#
+	 * @override
+	 */
+	this.width = texture.width;
+	/**
+	 * The height of the NineSlicePlane, setting this will actually modify the vertices and UV's of this plane
+	 *
+	 * @member {number}
+	 * @memberof PIXI.NineSlicePlane#
+	 * @override
+	 */
+	this.height = texture.height;
+
+	uvs[2] = uvs[10] = uvs[18] = uvs[26] = this._uvw * leftWidth;
+	uvs[4] = uvs[12] = uvs[20] = uvs[28] = 1 - this._uvw * rightWidth;
+	uvs[9] = uvs[11] = uvs[13] = uvs[15] = this._uvh * topHeight;
+	uvs[17] = uvs[19] = uvs[21] = uvs[23] = 1 - this._uvh * bottomHeight;
+
+	/**
+	 * The width of the left column (a)
+	 *
+	 * @member {number}
+	 */
+	this.leftWidth = leftWidth || DEFAULT_BORDER_SIZE;
```

Comment:
If `leftWidth` is `0` then this will always default to 10. This should support 3 or 6 slice scaling.